Bottom: 3f518c50d8354809224e4ab4b34e2a670885a71a
Top:    d57e3afd8921c3acb0cf93ffb59125d6e9477c5a
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-12-18 17:52:34 +0000

Refresh of hid-common-hidnogui-c-plug-lea

---

diff --git a/src/hid/common/hidnogui.c b/src/hid/common/hidnogui.c
index 9d88086..b08d90d 100644
--- a/src/hid/common/hidnogui.c
+++ b/src/hid/common/hidnogui.c
@@ -381,7 +381,7 @@ nogui_prompt_for (const char *msg, const char *default_string)
   if (answer == NULL)
     return strdup ((default_string != NULL) ? default_string : "");
   else
-    return strdup (answer);
+    return answer;
 }
 
 /* FIXME - this could use some enhancement to actually use the other
@@ -402,7 +402,7 @@ nogui_fileselect (const char *title, const char *descr,
   if (answer == NULL)
     return (default_file != NULL) ? strdup (default_file) : NULL;
   else
-    return strdup (answer);
+    return answer;
 }
 
 static int
